### ISSUE TITLE
#n1ql.bucket returns the collection name instead of bucket name

### ISSUE THREAD

**Issue by rbleuse** [spring-projects/spring-data-couchbase#1536]
Hello,

I'm using the 4.4.2 version
Running a query with custom n1ql hints fails as the bucket is wrongly replaced by the collection name.

Configuration :

```kt
@Configuration
@EnableReactiveCouchbaseRepositories
class CouchbaseConfig(
    private val properties: CouchbaseProperties
) : AbstractCouchbaseConfiguration() {
    override fun getConnectionString(): String {
        return properties.connectionString
    }

    override fun getUserName(): String {
        return properties.username
    }

    override fun getPassword(): String {
        return properties.password
    }

    override fun getBucketName(): String {
        return "test"
    }
}
```

Repository :

```kt
interface ScheduleRepository : ReactiveCouchbaseRepository<Schedule, String> {
    @Query("#{#n1ql.selectEntity} WHERE #{#n1ql.filter}")
    fun findByN1ql(): Flux<Schedule>

    @Query("SELECT c.* FROM #{#n1ql.bucket}.#{#n1ql.scope}.#{#n1ql.collection} c WHERE c.#{#n1ql.filter}")
    fun findByN1qlCustom(): Flux<Schedule>
}
```

Running the `findByN1ql` query runs fine, n1ql.selectEntity fetches data in the correct bucket/scope/collection

However, running the custom query `findByN1qlCustom` which specifies bucket/scope/collection doesn't work :

```
com.couchbase.client.core.error.IndexFailureException: The server reported an issue with the underlying index {"completed":true,"coreId":"0xe4295ea200000001","errors":[{"code":12003,"message":"Keyspace not found in CB datastore: default:schedule - cause: No bucket named schedule","retry":false}],"httpStatus":500,"idempotent":false,"lastDispatchedFrom":"127.0.0.1:55070","lastDispatchedTo":"127.0.0.1:8093","requestId":179,"requestType":"QueryRequest","retried":0,"service":{"bucket":"test","operationId":"d319c5bb-28b3-4b22-a38c-2be2d0a3a290","scope":"dev","statement":"SELECT c.* FROM `schedule`.`dev`.`schedule` c WHERE `_class` = \"com.rbleuse.spring.reactive.couchbase.model.Schedule\"","type":"query"},"timeoutMs":75000,"timings":{"dispatchMicros":27475,"totalDispatchMicros":27475,"totalMicros":33870}}
```

No bucket named `schedule`

**Comment by rbleuse** [comment]
I did a bit of investigation, and the issue seems to come from here : 

https://github.com/spring-projects/spring-data-couchbase/blob/4.4.x/src/main/java/org/springframework/data/couchbase/repository/query/StringBasedN1qlQueryParser.java#L158-L159

bucketName = test
collection = schedule

But because collection is not null, the `collection` is passed to that method instead of the `bucketName`

**Comment by mikereiche** [comment]
Instead use :

  `@Query("SELECT c.* FROM #{#n1ql.bucket} c WHERE c.#{#n1ql.filter}")`
  
  In 4.x, it works like that because when collection support was added, there was a desire that existing @Query methods written for buckets would also work for collections simply by configuring to use scopes and collections.  (and when there is a non-default scope, n1ql.bucket gives the collection name, and query option of  query_context of bucket.collection is added to the REST call to the couchbase server.)
  
When n1ql.scope and n1ql.collection were implemented in #1445, n1ql.bucket had to keep giving the collection as changing it would be a breaking change (backing out that change is mentioned in the issue).  So in 5.x, bucket is always bucket, scope is always scope and collection is always collection.  But in 4.x n1ql.bucket is bucketname when the scope/collection are _default (i.e. collections are not used), and n1ql.bucket is collectionname when the scope/collection are not _default. 

  The long story is in https://github.com/spring-projects/spring-data-couchbase/issues/1445

**Comment by rbleuse** [comment]
Hi @mikereiche thank you for the explanation, indeed that's working with your solution.

Closing this issue !
